src/main/java/com/google/sps/servlets/RecipeDetailsServlet.java

```diff
@@ -26,14 +35,42 @@
 
 @WebServlet("/recipe")
 public class RecipeDetailsServlet extends HttpServlet {
-  /*
-  * doGet receives the request and returns the message sent as parameter 
-  */
+
+  /**
+   * doGet receives the request and returns the message sent as parameter
+   **/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
     throws ServletException, IOException {
-    request.setAttribute("messages", request.getParameter("id"));
+    try {
+      String idRecipe = request.getParameter("id");
+
+      // get the recipe entity
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      long id = Long.parseLong(idRecipe);
+      Entity recipeEntity = datastore.get(KeyFactory.createKey("Recipe", id));
+
+      // send all the parameters to the request
+
+      request.setAttribute("title", recipeEntity.getProperty("title"));
+      request.setAttribute("author", recipeEntity.getProperty("author"));
+      request.setAttribute("imgURL", recipeEntity.getProperty("imgURL"));
+      request.setAttribute(
+        "difficulty",
+        recipeEntity.getProperty("difficulty")
+      );
+      request.setAttribute("prepTime", recipeEntity.getProperty("prep_time"));
+      request.setAttribute("cookTime", recipeEntity.getProperty("cook_time"));
+      request.setAttribute(
+        "ingredients",
+        recipeEntity.getProperty("ingredients")
+      );
+      request.setAttribute("steps", recipeEntity.getProperty("stepList"));
 
-    request.getRequestDispatcher("/recipe.jsp").forward(request, response);
+      request.getRequestDispatcher("/recipe.jsp").forward(request, response);
+    } catch (EntityNotFoundException e) {
+      request.setAttribute("error", 1);
+      request.getRequestDispatcher("/recipe.jsp").forward(request, response);
+    }
```

Comment:
Please enclose in the try block only the calls than can actually throw those exceptions.